java/flight/src/main/java/org/apache/arrow/flight/FlightDescriptor.java

```diff
@@ -160,7 +161,8 @@ public boolean equals(Object obj) {
       if (other.cmd != null) {
         return false;
       }
-    } else if (!Arrays.equals(cmd, other.cmd)) {
+    } else if (ByteFunctionHelpers.equal(
```

Comment:
This isn't performance sensitive. Please don't use non-idiomatic code for performance purposes where performance doesn't matter.

Reply:
Reverted. Thanks.